docs/reference/search/uri-request.asciidoc

```diff
@@ -101,10 +101,12 @@ is important).
 |`track_scores` |When sorting, set to `true` in order to still track
 scores and return them as part of each hit.
 
-|`track_total_hits` |Set to `false` in order to disable the tracking
+|`track_total_hits` |Defaults to true. Set to `false` in order to disable the tracking
 of the total number of hits that match the query.
-(see <<index-modules-index-sorting,_Index Sorting_>> for more details).
-Defaults to true.
+It also accepts an integer which in this case represents the number of
```

Comment:
@jimczi , this is problematic in strongly-typed languages, since the `track_total_hits` parameter is defined as a `boolean`, and eg. the Go test suite fails with `cannot use 4 (type int) as type bool in field value`:

https://github.com/elastic/elasticsearch/blob/b34e7d4f192c089c202465ecbe7aed730aaa1a02/rest-api-spec/src/main/resources/rest-api-spec/test/search/220_total_hits_object.yml#L117

I understand the motivation here, but we should at least revisit the parameter definitions, and support something like `"type" : ["boolean","number]"`, so the code generators can make decisions here.